tests/Pipeline/PipelineTest.php

```diff
@@ -27,6 +27,21 @@ public function testPipelineBasicUsage()
         unset($_SERVER['__test.pipe.two']);
     }
 
+    public function testPipelineUsageWithObjects()
+    {
+        $result = (new Pipeline(new Illuminate\Container\Container))
+            ->send('foo')
+            ->through([new PipelineTestPipeOne])
+            ->then(function ($piped) {
+                return $piped;
+            });
+
+        $this->assertEquals('foo', $result);
+        $this->assertEquals('foo', $_SERVER['__test.pipe.one']);
+
+        unset($_SERVER['__test.pipe.one']);
```

Comment:
@GrahamCampbell there is no exception catching in most of unit tests, moreover the previous one (line 17) will throw uncaught ReflectionException in case class does not exist, so it will be screwed up even without my test :smiley: 
```
Time: 1.93 seconds, Memory: 64.00Mb

There were 3 errors:

1) PipelineTest::testPipelineBasicUsage
ReflectionException: Class PipelineTestPipeOne does not exist

/home/likewise-open/FRAMGIA/roman.kinyakin/Code/framework/src/Illuminate/Container/Container.php:738
/home/likewise-open/FRAMGIA/roman.kinyakin/Code/framework/src/Illuminate/Container/Container.php:633
/home/likewise-open/FRAMGIA/roman.kinyakin/Code/framework/src/Illuminate/Pipeline/Pipeline.php:122
/home/likewise-open/FRAMGIA/roman.kinyakin/Code/framework/src/Illuminate/Pipeline/Pipeline.php:102
/home/likewise-open/FRAMGIA/roman.kinyakin/Code/framework/tests/Pipeline/PipelineTest.php:20

2) PipelineTest::testPipelineUsageWithObjects
Error: Class 'PipelineTestPipeOne' not found

/home/likewise-open/FRAMGIA/roman.kinyakin/Code/framework/tests/Pipeline/PipelineTest.php:34

3) PipelineTest::testPipelineViaChangesTheMethodBeingCalledOnThePipes
ReflectionException: Class PipelineTestPipeOne does not exist

/home/likewise-open/FRAMGIA/roman.kinyakin/Code/framework/src/Illuminate/Container/Container.php:738
/home/likewise-open/FRAMGIA/roman.kinyakin/Code/framework/src/Illuminate/Container/Container.php:633
/home/likewise-open/FRAMGIA/roman.kinyakin/Code/framework/src/Illuminate/Pipeline/Pipeline.php:122
/home/likewise-open/FRAMGIA/roman.kinyakin/Code/framework/src/Illuminate/Pipeline/Pipeline.php:102
/home/likewise-open/FRAMGIA/roman.kinyakin/Code/framework/tests/Pipeline/PipelineTest.php:70

--

```